### PR TITLE
Npm module

### PR DESCRIPTION
**Pull request by justingeeslin**
- The ability for a user to npm install this package.
- In npm style, a project can have playcanvas (and ammo) as it dependencies
- The simple, standard "npm run build" now builds the project.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,3 @@
 *.DS_Store
-/build/output
```
Comment (Maksims) [inline .gitignore]: Built output should not be included into repository. When installing using npm it probably should build it as part of installation.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "playcanvas",
+  "version": "1.0.0",
```
Comment (Maksims) [inline package.json]: version number is already mentioned below, this leads to duplicate key in JSON.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{
```
Comment (Maksims) [inline package.json]: Please use 4 spaces intention as by our coding standards.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/playcanvas/engine"
+  },
+  "engines": {
+      "node": ">= 0.6.12"
+  },
+  "author": "PlayCanvas <support@playcanvas.com>",
```
Comment (Maksims) [inline package.json]: Please move keys for better readability to such order:
name, version, author, homepage, description, license, main, bugs, repository, dependencies, devDependencies, directories, engines, scripts

File: package.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  "name": "playcanvas",
+  "version": "1.0.0",
+  "description": "JavaScript game engine built on WebGL and WebVR. https://playcanvas.com/",
+  "main": "build/output/playcanvas-latest.min.js",
+  "directories": {
+    "example": "examples",
+    "test": "tests"
+  },
+  "scripts": {
+    "build": "cd build && node build.js -l 1 -o output/playcanvas-latest.min.js && node build.js -l 0 -o output/playcanvas-latest.js && node build.js -l 0 -d -o output/playcanvas-latest.dbg.js",
```
Comment (Maksims) [inline package.json]: There is also `-p` for profiler define option available.

File: src/core/core.js [hunk 6]
```diff
@@ -170,3 +170,7 @@ var pc = {
 if (typeof (exports) !== 'undefined') {
     exports.pc = pc;
 }
+if (typeof module === 'undefined') {
+  var module = {}
```
Comment (Maksims) [inline src/core/core.js]: We use 4 space indention.

File: package.json [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/playcanvas/engine"
+  },
+  "engines": {
+      "node": ">= 0.6.12"
+  },
+  "author": "PlayCanvas <support@playcanvas.com>",
+  "license": "ISC",
+  "version": "0.206.0-dev",
```
Comment (Maksims) [inline package.json]: The reason why package.json was generated as part of build process, is that it updated the `version` based on content of VERSION file, and it is still part of our deployment script, which ensures we keep version number in track all around.

File: package.json [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    "url": "https://github.com/playcanvas/engine/issues"
+  },
+  "homepage": "https://playcanvas.com",
+  "devDependencies": {
+    "fs-extra": "^1.0.0",
+    "google-closure-compiler": "^20161201.0.0",
+    "preprocessor": "^1.4.0"
+  },
+  "dependencies": {
+    "ammo.js": "https://github.com/kripken/ammo.js"
```
Comment (Maksims) [inline package.json]: Our engine actually depends on modified ammo.js version which is here: https://github.com/playcanvas/ammo.js

**Comment by willeastcott** [comment]
This hasn't been updated in some time. What's the plan with this?

**Comment by Maksims** [comment]
Closing due to lack of activity.

**Pull request closed without merging**